spec/factories/notes.rb

```diff
@@ -0,0 +1,18 @@
+FactoryBot.define do
+  factory :note do
+    title { Faker::Marketing.buzzwords }
+    content { Faker::Lorem.paragraph_by_chars(number: 256) }
+
+    trait :on_student do
+      association :entity, factory: :student
+    end
+
+    trait :on_course do
+      association :entity, factory: :course
+    end
+
+    trait :on_institute do
+      association :entity, factory: :institute
+    end
```

Comment:
I generally feel it's good for traits to be exclusive of one another so you can apply multiple of them without them affecting one another, so in this case doing `create(:note, :on_student, :on_course)` would note give the desired affect or make sense.
I suggest doing sub factories instead for this so the traits cannot be chained. 

```suggestion
    factory :student_note do
      association :entity, factory: :student
    end

    factory :course_note do
      association :entity, factory: :course
    end

    factory :institute_note do
      association :entity, factory: :institute
    end
```